### ISSUE TITLE
Remap errors out after remove rows

### ISSUE THREAD

**Issue by monique2208** [hed-standard/hed-python#927]
When the remap operation is called after the remove rows operation the following error occurs

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\.virtualenvs\hedtools\Scripts\run_remodel.exe\__main__.py", line 7, in <module>
  File "C:\Users\.virtualenvs\hedtools\Lib\site-packages\hed\tools\remodeling\cli\run_remodel.py", line 218, in main
    run_direct_ops(dispatch, args, files)
  File "C:\Users\.virtualenvs\hedtools\Lib\site-packages\hed\tools\remodeling\cli\run_remodel.py", line 185, in run_direct_ops
    df = dispatch.run_operations(file_path, verbose=args.verbose, sidecar=sidecar)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\\.virtualenvs\hedtools\Lib\site-packages\hed\tools\remodeling\dispatcher.py", line 153, in run_operations
    df = operation.do_op(self, df, file_path, sidecar=sidecar)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\b.virtualenvs\hedtools\Lib\site-packages\hed\tools\remodeling\operations\remap_columns_op.py", line 118, in do_op
    df_new, missing = self.key_map.remap(df1)
                      ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\.virtualenvs\hedtools\Lib\site-packages\hed\tools\analysis\key_map.py", line 113, in remap
    missing_indices = self._remap(df_new)
                      ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\.virtualenvs\hedtools\Lib\site-packages\hed\tools\analysis\key_map.py", line 135, in _remap
    df.iloc[index] = row
    ~~~~~~~^^^^^^^
  File "C:\Users\.virtualenvs\hedtools\Lib\site-packages\pandas\core\indexing.py", line 908, in __setitem__
    self._has_valid_setitem_indexer(key)
  File "C:\Users\.virtualenvs\hedtools\Lib\site-packages\pandas\core\indexing.py", line 1646, in _has_valid_setitem_indexer
    raise IndexError("iloc cannot enlarge its target object")
IndexError: iloc cannot enlarge its target object
```
It seems the iterrows returns the index labels, which are not the same as the integer-location based indexes anymore after rows have been removed. iloc expects an integer-location based index which at some point falls outside of the dataframe's range.

**Comment by VisLab** [comment]
@IanCa  can you please take a look at this.

**Comment by IanCa** [comment]
Is there a good test case?

I don't know this part of the code code much at all.

**Comment by monique2208** [comment]
I don' t have the original files but with this example I get the same result

I could try to make a pull request as well. I think I know what the issue is. 

Pandas has two ways of indexing dataframes. Each dataframe has a kind of index 'column', and then there is just indexing based on location. When we load event files, pandas autmatically adds this index 'column', and the location index and index label are exactly the same. However, as soon as wel remove rows, this is not the case anymore. 

I think we should actually fix two things:

It would be good for us to reset the index column after we remove rows. I think this would just be good practice. 

However, I think the code where the error is thrown should also be adapted. Because right now it uses one function 'iterrows', which explicitly returns index labels, and then the function 'iloc' which expects these  integer-location based index. This only ever works if these are the same, which is not guarenteed. To make the code more robust, might want to turn the index provided by iterrows into the integer location index using 'df.index.get_loc'. 

[test_error.zip](https://github.com/user-attachments/files/15807041/test_error.zip)

**Comment by VisLab** [comment]
Fixed in #952
